back/engines/commercial/texting/app/models/texting/campaign.rb

```diff
@@ -10,10 +10,46 @@
 #  created_at    :datetime         not null
 #  updated_at    :datetime         not null
 #
+
 module Texting
   class Campaign < ApplicationRecord
+    enum status: {
+      draft: 'draft',
+      sending: 'sending',
+      sent: 'sent'
+    }.freeze
+
+    before_validation :format_phone_numbers
+
     validates :phone_numbers, :message, :status, presence: true
+    validate :validate_phone_numbers
+
+    after_initialize do
+      self.status ||= self.class.statuses.fetch(:draft)
+    end
+
+    def self.segments_count
```

Comment:
Looking at the code, it looks like we only count SMSes that were sent this month, but from the method name, I would think it would be the count of all times. Could it make sense to change the name to something like `segments_this_month_count`?